include/yaml-cpp/fp_to_string.h

```diff
@@ -0,0 +1,206 @@
+// SPDX-FileCopyrightText: 2024 Simon Gene Gottlieb
+// SPDX-License-Identifier: MIT
+
+#ifndef YAML_H_FP_TO_STRING
+#define YAML_H_FP_TO_STRING
+
+#include "contrib/dragonbox.h"
+
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <sstream>
+#include <tuple>
+
+namespace YAML {
+namespace detail {
+namespace fp_formatting {
+
+/**
+ * Converts a integer into its ASCII digits.
+ *
+ * @param begin/end - a buffer, must be at least 20bytes long
+ * @param value     - input value
+ * @param width     - minimum number of digits, fill with '0' to the left. Must be equal or smaller than the buffer size.
+ * @return          - number of digits filled into the buffer.
+ *
+ * Example:
+ * std::array<char, 20> buffer;
+ * auto ct = convertToChars(buffer.begin(), buffer.end(), 23, 3);
+ * assert(ct = 3);
+ * assert(buffer[0] == '0');
+ * assert(buffer[1] == '2');
+ * assert(buffer[2] == '3');
+ */
+inline auto convertToChars(char* begin, char* end, size_t value, int width=1) -> int {
```

Comment:
It looks like the rest of the yaml-cpp codebase uses PascalCase for function names. I think we should make this consistent with that style.